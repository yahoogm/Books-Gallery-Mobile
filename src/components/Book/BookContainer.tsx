import Book from './Book';
import {useAppSelector} from '../../hooks/useRedux';
import {detailBookSelector} from '../../redux/book/bookSelector';
import {modifiedName} from '../../../utils/const/const';
import {BookContainerProps} from './types';

const BookContainer: React.FC<BookContainerProps> = ({
  refReadBook,
  setShowBook,
}) => {
  const detailBook = useAppSelector(detailBookSelector);

  const authorsName = modifiedName(detailBook.volumeInfo.authors);
  const categories = modifiedName(detailBook.volumeInfo.categories);

  const uriDetailBook =
    detailBook.volumeInfo.imageLinks &&
    detailBook.volumeInfo.imageLinks.thumbnail
      ? detailBook.volumeInfo.imageLinks.thumbnail
      : 'https://plus.unsplash.com/premium_photo-1698084059435-a50ddfd69303?q=80&w=1850&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D';
  return (
    <Book
      setShowBook={setShowBook}
      refReadBook={refReadBook}
      detailBook={detailBook}
      categories={categories}
      uriDetailBook={uriDetailBook}
      authorsName={authorsName}
    />
  );
};

export default BookContainer;
