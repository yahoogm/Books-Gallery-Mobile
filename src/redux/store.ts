import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit';
import {Reducers} from './reducer/reducer';

export const Store = configureStore({
  reducer: Reducers,
});

export type RootState = ReturnType<typeof Store.getState>;
export type AppDispatch = typeof Store.dispatch;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
