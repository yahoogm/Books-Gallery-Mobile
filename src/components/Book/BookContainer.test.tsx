import React, {createRef} from 'react';
import renderer from 'react-test-renderer';
import BookContainer from './BookContainer';
import {Provider} from 'react-redux';
import {Store} from '../../redux/store';

describe('BookContainerComponent', () => {
  it('renders corretly when given props', () => {
    const mockSetShowBook = jest.fn();
    const refReadBook = createRef<any>();
    const tree = renderer
      .create(
        <Provider store={Store}>
          <BookContainer
            refReadBook={refReadBook}
            setShowBook={mockSetShowBook}
          />
          ,
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
