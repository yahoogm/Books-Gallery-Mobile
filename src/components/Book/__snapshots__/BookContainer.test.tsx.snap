// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookContainerComponent renders corretly when given props 1`] = `
<View
  data-style=""
  dataSet={
    {
      "style": "",
    }
  }
  style={
    [
      undefined,
      {},
      {
        "alignItems": "center",
        "display": "flex",
        "padding": 20,
      },
      undefined,
    ]
  }
>
  <Image
    alt="title"
    aria-label="title"
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    h={340}
    role="img"
    source={
      {
        "uri": "https://plus.unsplash.com/premium_photo-1698084059435-a50ddfd69303?q=80&w=1850&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
      }
    }
    style={
      [
        undefined,
        {},
        {
          "borderRadius": 10,
          "width": 250,
        },
        undefined,
      ]
    }
  />
  <View
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    style={
      [
        undefined,
        {},
        {
          "display": "flex",
          "flexDirection": "row",
          "gap": 10,
          "marginTop": 10,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      data-style=""
      dataSet={
        {
          "active": "false",
          "disabled": "false",
          "focus": "false",
          "focusVisible": "false",
          "hover": "false",
          "style": "",
        }
      }
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      style={
        [
          undefined,
          {},
          undefined,
        ]
      }
    >
      <Text
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        style={
          [
            undefined,
            {},
            undefined,
            {},
            undefined,
          ]
        }
      >
        Baca Buku
      </Text>
    </View>
  </View>
  <View
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    style={
      [
        undefined,
        {},
        {
          "alignItems": "center",
          "marginTop": 10,
        },
        undefined,
      ]
    }
  >
    <View
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      style={
        [
          undefined,
          {},
          {
            "alignItems": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        accessibilityRole="header"
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        size="md"
        style={
          {
            "fontSize": 32,
            "fontWeight": "bold",
            "marginVertical": 42.56,
          }
        }
      >
        Author:
      </Text>
      <Text
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        style={
          [
            undefined,
            {},
            undefined,
          ]
        }
      />
    </View>
    <View
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      style={
        [
          undefined,
          {},
          {
            "alignItems": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        accessibilityRole="header"
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        size="md"
        style={
          {
            "fontSize": 32,
            "fontWeight": "bold",
            "marginVertical": 42.56,
          }
        }
      >
        Publisher:
      </Text>
      <Text
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        style={
          [
            undefined,
            {},
            undefined,
          ]
        }
      />
    </View>
    <View
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      style={
        [
          undefined,
          {},
          {
            "alignItems": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        accessibilityRole="header"
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        size="md"
        style={
          {
            "fontSize": 32,
            "fontWeight": "bold",
            "marginVertical": 42.56,
          }
        }
      >
        Published Date:
      </Text>
      <Text
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        style={
          [
            undefined,
            {},
            undefined,
          ]
        }
      />
    </View>
    <View
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      style={
        [
          undefined,
          {},
          {
            "alignItems": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        accessibilityRole="header"
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        size="md"
        style={
          {
            "fontSize": 32,
            "fontWeight": "bold",
            "marginVertical": 42.56,
          }
        }
      >
        Page Count:
      </Text>
      <Text
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        style={
          [
            undefined,
            {},
            undefined,
          ]
        }
      >
        0
      </Text>
    </View>
    <View
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      style={
        [
          undefined,
          {},
          {
            "alignItems": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        accessibilityRole="header"
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        size="md"
        style={
          {
            "fontSize": 32,
            "fontWeight": "bold",
            "marginVertical": 42.56,
          }
        }
      >
        Category:
      </Text>
      <Text
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        style={
          [
            undefined,
            {},
            undefined,
          ]
        }
      />
    </View>
    <View
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      style={
        [
          undefined,
          {},
          {
            "alignItems": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        accessibilityRole="header"
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        size="md"
        style={
          {
            "fontSize": 32,
            "fontWeight": "bold",
            "marginVertical": 42.56,
          }
        }
      >
        Description:
      </Text>
      <Text
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        style={
          [
            undefined,
            {},
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
